packages/api/tests/db-data-helpers/create-test-data.js

```diff
@@ -0,0 +1,178 @@
+'use strict';
+
+const { randomString, randomId } = require('@cumulus/common/test-utils');
+const { s3PutObject } = require('@cumulus/aws-client/S3');
+const {
+  FilePgModel,
+  GranulePgModel,
+  CollectionPgModel,
+} = require('@cumulus/db');
+
+const { constructCollectionId } = require('@cumulus/message/Collections');
+
+// PG mock data factories
+const {
+  fakeGranuleRecordFactory,
+  fakeCollectionRecordFactory,
+} = require('@cumulus/db/dist/test-utils');
+
+const {
+  createS3Buckets,
+} = require('@cumulus/aws-client/S3');
+
+// Dynamo mock data factories
+const {
+  fakeGranuleFactoryV2,
+  fakeCollectionFactory,
+} = require('../../lib/testUtils');
+
+const models = require('../../models');
+
+/**
+ * Helper for creating a granule, a parent collection,
+ * and files belonging to that granule (in S3 and PG)
+ * @param {Knex} dbClient - Knex client
+ * @param {number} collectionId - collectionId for the granule's parent collection
+ * @param {number} collectionCumulusId - cumulus_id for the granule's parent collection
+ * @param {boolean} published - if the granule should be marked published to CMR
+ * @returns {Object} fake granule object
+ */
+async function createGranuleAndFiles({
```

Comment:
Moving this to a separate file because we're using it in 3 places. This could definitely be made more generic. Right now it does a lot. If this proves valuable it could be split up to handle things a little nicer.